docs/architecture/decisions/0014-dynamic-datasource-registration.md

```diff
@@ -0,0 +1,106 @@
+# 0014. Dynamic Data Source Registration
+
+## Status
+
+Accepted
+
+## Context
+
+Decouple the data sources from the SDK into their own NuGet packages and create a mechanism to allow auto discovery and dynamic registration. This will allow for new data sources to be plugged in with minimum configuration and dependency on the existing code base. 
+
+## Decision
+
+- Decoupling from SDK
+
+Each data source can be built externally with no dependency on the CarbonAware SDK. This helps achieve single responsibility principle and makes the system extensible.
```

Comment:
It depends how data source would be interact with the SDK. It could be done using hard interface definitions, or also driven by manifest(s) that can help with the discoverability and how to interact with both entities.